prov/efa/docs/efa_rdm_protocol_v4.md

```diff
@@ -0,0 +1,1318 @@
+a LONGREAD_RTM packet's mandatory header is listed in table 4.1
+
+Table: 4.1 the binary format of a LONGREAD_RTM packet's mandatory header
+
+| Name | Length (bytes) | type | C language type | Note |
+|-|-|-|-|-|
+| `type`           | 1 | integer | `uint8_t`  | part of base header |
+| `version`        | 1 | integer | `uint8_t`  | part of base header|
+| `flags`          | 2 | integer | `uint16_t` | part of base header |
+| `msg_id`         | 4 | integer | `uint32_t` | message ID |
+| `msg_length`     | 4 | integer | `uint64_t` | total length of the message |
+| `send_id`        | 4 | integer | `uint32_t` | ID of the receive operation  |
+| `read_iov_count` | 4 | integer | `uint32_t` | number of iov to read |
+| `read_iov`       | `read_iov_count` * 24 | array | `efa_rma_iov[]` | read iov information |
+
+In LONGREAD_RTM packet mandatory header, there are two new fields `read_iov_count` and `read_iov`.
+They are different from the `rma_iov_count` and `rma_iov` fields introduce in chapter 3.
+`rma_iov_count` and `rma_iov` is provided by application when it called libfabric's one-sided API.
+In long-read message sub-protocol, sender need to construct the `read_iov_count` and `read_iov`
+from the send buffer provided by application. For that, sender need to make sure the send buffer
+is registered with EFA device and fill the registration key in `read_iov`. There are two ways
+to achieve that: first, if the buffer has already been registered with device, application will provide
+a memory descriptor along with the sender buffer, registration key can be extracted from the descriptor;
+second, if the buffer has not been registered with EFA device, sender need to register the buffer,
+and can get the key from the registration. Note because memory registration is a limited resource,
+it is possible for memory registration to fail and sender need to be able to handle the case.
+
+Upon receiving a long-read RTM, the receiver will use RDMA read to copy data from application's
+send buffer to application's receive buffer (avoiding copy). That is why this protocol is
+sometime referred as zero-copy.
+
+After all read is finished, the receiver will send an EOR packet to the sender to notify it
+the work is done.
+
+The binary format of the EOR packet is listed in table 4.2
+
+Table: 4.2 the format of an EOR packet
+
+| Name | Length (bytes) | type | C language type | Note |
+|-|-|-|-|-|
+| `type`           | 1 | integer | `uint8_t`  | part of base header |
+| `version`        | 1 | integer | `uint8_t`  | part of base header|
+| `flags`          | 2 | integer | `uint16_t` | part of base header |
+| `send_id`        | 4 | integer | `uint32_t` | ID of the send operation |
+| `recv_id`        | 4 | integer | `uint32_t` | ID of the receive operation |
+| `connid`         | 4 | integer | `uint32_t` | optional sender connection ID, see section 4.4 for more detail|
+
+#### emulated long-read write sub-protocol
```

Comment:
Is there a better name for this? It's kind of hard to parse although I know what you're talking about here. Maybe "emulated long RDMA write via RDMA read"?